lib/builder.dart

```diff
@@ -54,30 +54,58 @@ import 'package:coap/coap.dart';
 /// the config file to contain only those entries that override the defaults.
 /// The file can't be empty, so version must as a minimum be present.
 class $className extends DefaultCoapConfig {
-${_generateDataScript(data)}}
+${_generateDataScript(data, '')}}
 """;
 
-String _generateDataScript(YamlMap data) {
+String _generateDataScript(YamlMap data, String prefix, {firstLine = true}) {
   final buff = StringBuffer();
-  for (final k in data.keys) {
-    buff.writeln('  @override');
-    if (data[k] is String) {
-      if (data[k] == 'true' || data[k] == 'false') {
-        buff.writeln('  bool get $k => ${data[k]};');
-        continue;
+
+  data.forEach((key, value) {
+    if (value is YamlMap) {
+      buff.write(_generateDataScript(value, key, firstLine: false));
+      return;
+    }
+    final variableName =
+        prefix.isEmpty ? key : prefix + key[0].toUpperCase() + key.substring(1);
+
+    if (variableName == 'dtlsBackend') {
```

Comment:
Should we use the `key` here (and for similar comparisons), or will these comparisons work even if a prefix is set?